standardized-output-composition.md

```diff
@@ -0,0 +1,291 @@
+  "type": "array",
+  "items": { "$ref": "#/$defs/point" },
+  "minItems": 3
+}
+
+[
+  {
+    "x": 2.5,
+    "y": 1.3,
+  },
+  {
+    "x": 1,
+    "z": 6.7
+  }
+]
+```
+
+This instance will fail validation and produce errors, but it's trivial to deduce examples for passing schemas that produce annotations.
+
+Specifically, the errors it will produce are:
+
+1. The second element in the `vertices` property is missing a `y` property.
+2. The second element in the `vertices` property has a disallowed `z` property.
+3. There are only two vertices, but three are required.
+
+Note that the wording in the error messages depicted in these examples is not a requirement of this specification.  Implementations SHOULD craft error messages tailored for their audience.
+
+### Pass/Fail
+
+In the simplest case, merely the boolean result for the `valid` valid property needs to be fulfilled.
+
+```json
+{
+  "valid": false
+}
+```
+
+Because no errors or annotations are returned with this format, it is RECOMMENDED that implementations use short-circuiting logic to return failure or success as soon as the outcome can be determined.  For example, if an `anyOf` keyword contains five sub-schemas, and the second one passes, there is no need to check the other three.  The logic can simply return with success.
+
+### Basic
+
+The **Basic** structure is a flat list of output units.
+
+```json
+{
+  "valid": false,
+  "errors": [
+    {
+      "keywordLocation": "#",
+      "instanceLocation": "#",
+      "error": "A subschema had errors."
+    },
+    {
+      "keywordLocation": "#/items/$ref",
+      "absoluteKeywordLocation":
+        "http://example.com/polygon#/definitions/point",
+      "instanceLocation": "#/1",
+      "error": "A subschema had errors."
+    },
+    {
+      "keywordLocation": "#/items/$ref/required",
+      "absoluteKeywordLocation":
+        "http://example.com/polygon#/definitions/point/required",
+      "instanceLocation": "#/1",
+      "keyword": "required",
+      "error": "Required property 'y' not found."
+    },
+    {
+      "keywordLocation": "#/items/$ref/additionalProperties",
+      "absoluteKeywordLocation":
+        "http://example.com/polygon#/definitions/point/additionalProperties",
+      "instanceLocation": "#/1/z",
+      "error": "Additional property 'z' found but was invalid."
+    },
+    {
+      "keywordLocation": "#/minItems",
+      "instanceLocation": "#",
+      "keyword": "minItems",
+      "error": "Expected at least 3 items but found 2"
+    }
+  ]
+}
+```
+
+### Detailed
+
+The **Detailed** structure is based on the schema and can be more readable for both humans and machines.  Having the structure organized this way makes associations between the errors more apparent.  For example, the fact that the missing `y` property and the extra `z` property both stem from the same location in the instance is not immediately obvious in the **Basic** structure.  In a hierarchy, the correllation is more easily identified.
```

Comment:
Is it possible to build the detailed structure given the basic structure? Just curious.

Reply:
I *think* so. It would take quite a bit more processing because you have to look at the keyword location properties to reconstruct the hierarchy, but I don't think it's impossible.